src/features/embalm/stepContent/hooks/useCreateSarcohpagus.ts

```diff
@@ -1,141 +1,105 @@
+import { ethers } from 'ethers';
+import { doubleHashShard, encrypt, readFileDataAsBase64 } from 'lib/utils/helpers';
+import { useCallback } from 'react';
 import { split } from 'shamirs-secret-sharing-ts';
 import { setIsUploading } from 'store/bundlr/actions';
 import {
-  setPayloadTxId,
-  setSelectedArchaeologists,
-  setShardsTxId,
-  setShards,
+  setShardPayloadData,
 } from 'store/embalm/actions';
 import { useDispatch, useSelector } from 'store/index';
-import { Archaeologist } from 'types/index';
-import { useBundlr } from './useBundlr';
+// import { useBundlr } from './useBundlr';
 import { useSubmitSarcophagus } from 'hooks/embalmerFacet';
-
-async function encryptShards(publicKeys: string[], payload: Uint8Array[]): Promise<Buffer[]> {
-  return Promise.all(publicKeys.map(async (key, i) => encrypt(key, Buffer.from(payload[i]))));
+import { ArchaeologistEncryptedShard } from 'types';
+import useArweaveService from 'hooks/useArweaveService';
+
+async function encryptShards(publicKeys: string[], payload: Uint8Array[]): Promise<ArchaeologistEncryptedShard[]> {
+  return Promise.all(publicKeys.map(async (publicKey, i) => ({
+    publicKey,
+    encryptedShard: ethers.utils.hexlify(await encrypt(publicKey, Buffer.from(payload[i]))),
+    unencryptedShardDoubleHash: doubleHashShard(payload[i]),
+  })));
 }
 
 export function useCreateSarcophagus() {
   const dispatch = useDispatch();
   const {
-    recipientState,
+    // recipientState,
     file,
     outerPublicKey,
     outerPrivateKey,
     selectedArchaeologists,
     payloadTxId,
     shardsTxId,
+    totalArchaeologists,
+    requiredArchaeologists,
   } = useSelector(x => x.embalmState);
   const { isUploading } = useSelector(x => x.bundlrState);
-  const { uploadFile } = useBundlr();
-  const { createSarcophagus } = useSubmitSarcophagus();
+  // const { uploadFile } = useBundlr();
+  const { uploadArweaveFile } = useArweaveService();
+  const { submitSarcophagus } = useSubmitSarcophagus();
 
   // TODO: Move this into its own hook and check all fields
-  const canCreateSarcophagus = recipientState.publicKey !== '' && !!outerPublicKey && !!file;
+  const canCreateSarcophagus = !!outerPublicKey && !!file;
+  // const canCreateSarcophagus = recipientState.publicKey !== '' && !!outerPublicKey && !!file; // TODO: Uncomment
 
-  const handleSubmit = useCallback(async () => {
+  const uploadToArweave = useCallback(async () => {
+    const emptyData = { encryptedShards: [], encryptedShardsTxId: '' };
     try {
       // Prepare the payload
-      if (recipientState.publicKey === '' || !file || !outerPublicKey) return;
+      if (!canCreateSarcophagus) return emptyData;
+
       const payload = await readFileDataAsBase64(file);
 
       // Step 1: Encrypt the inner layer
-      const encryptedInnerLayer = await encrypt(recipientState.publicKey, payload);
+      // const encryptedInnerLayer = await encrypt(recipientState.publicKey, payload); // TODO: Uncomment
```

Comment:
Could we clean up some of these commented out lines or are we expecting to use them soon?

Reply:
Yeah. The line below this commented out code is technically wrong and should be replaced with this line as soon as the create sarco flow can be used properly in-app, which is soon.

There's a bunch of other similarly commented out lines that I've marked with TODOs

Reply:
@DarksightKellar -- it may help if the TODO had a little more detail about when this line should be uncommented, i.e. what needs to happen first (same with the other commented out lines in this file)

I realize this was probably more of a note to yourself since you may be handling these changes, but would help clarify for the reviewer.

Reply:
Good point